Making ratelimits configuration keys backwards compatible

oleurud opened the pull request:
Fix: https://github.com/CartoDB/CartoDB-SQL-API/issues/484

dgaubert commented:
Do you need a CR here?

enekid commented on app/server.js:
```diff
@@ -139,7 +139,7 @@ function App(statsClient) {
     const userLimitsServiceOptions = {
         limits: {
-            rateLimitsEnabled: global.settings.ratelimits.rateLimitsEnabled
+            rateLimitsEnabled: (global.settings.ratelimits && global.settings.ratelimits.rateLimitsEnabled) || false
```
another way:
```javascript
rateLimitsEnabled: Boolean(global.settings.ratelimits && global.settings.ratelimits.rateLimitsEnabled)
```

oleurud commented:
I have changed my point of view to solve it. Initially, my idea was adding more controls avoiding the `Cannot read property 'rateLimitsEnabled' of undefined` adding more checks. But I don't like it because the checks (although simples) were in the middle of the rate limit process. 

I have added a configuration method ensuring the correct parameters and It makes the control explicit. I would prefer something like server_options of Windshaft-carto, but it is not bad

oleurud commented on app/services/user_limits.js:
```diff
@@ -11,6 +22,13 @@ class UserLimits {
+    static configure() {
+        // default rate limits
+        if(!global.settings.ratelimits) {
+            global.settings.ratelimits = DEFAULT_RATE_LIMITS_OPTIONS;
```
:+1:

oleurud commented on app/services/user_limits.js:
```diff
@@ -11,6 +22,13 @@ class UserLimits {
         this.preprareRateLimit();
     }
 
+    static configure() {
```
Initially, I have added it to server.js (we can't add it to app.js because the tests don't use it). But finally, I have preferred to write it in the related code (rate limits are managed in user_limits.js), in order to maintain server.js small. 

But I don't have a preference, I don't really like in any case.

enekid commented on app/server.js:
```diff
@@ -66,6 +66,19 @@ function App(statsClient) {
+            rateLimitsEnabled: false,
+            endpoints: {
+                query: false,
+                query_format: false,
```
Just for the sake of pickiness: Why are those properties named with `_` as word separators instead of the standard camelCase way?

oleurud replied:
Because they are used in Ruby and Redis too. We find a consensus between the parts, and this was the result.
